### DISCUSSION TITLE
Can ipxe boot from password/mfa protected website (e.g. AWS EC2)

### DISCUSSION THREAD

**Discussion by FightingLemming** [ipxe/ipxe#479] [Q&A]
Greetings! 
I'm kicking around the idea of building a deployment server in AWS that would be publicly available. Only problem is... publicly available means security risk. Does ipxe have a way to interact with a username, password, and MFA so I can protect that public facing web address?

Please and thank you! 🙏🙏🙏

![were-not-worthy-waynes-world](https://user-images.githubusercontent.com/59526591/134591975-cc0d37fd-52e5-4b5b-980e-20ac99835e07.gif)

**Comment by NiKiZe** [comment]
Consider iPXE being a console app. You could do this by having serverside handling. Show URL in iPXE that user has to visit from their device that then does the auth, iPXE in the meantime does long polling, or retrys.

Other than this iPXE has a login prompt, MFA prompt could be done with scripting.
